### PR TITLE
Avoid finding source packages in dnf_context_install()

### PR DESCRIPTION
**Pull request by cgwalters**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1565647

I think this was a subtle regression introduced from
porting `dnf_context_install()` to match the dnf behavior.
That was in: https://github.com/rpm-software-management/libdnf/pull/204

Basically we started looking at source packages particularly when matching
`Provides`.

I'm really uncertain if there are actually any use cases at all
for finding source packages from these APIs, but I went the
maximally conservative route and added booleans.

### CODE REVIEW

**Comment by cgwalters** [comment]
Use: https://github.com/projectatomic/rpm-ostree/pull/1349

File: libdnf/hy-subject.cpp [hunk 1]
```diff
@@ -349,13 +360,17 @@ hy_subject_get_best_solution(HySubject subject, DnfSack *sack, HyForm *forms, Hy
...
+                         bool with_src)
 {
     HyNevra nevra = NULL;
     HyQuery query = hy_subject_get_best_solution(subject, sack, forms, &nevra, FALSE, TRUE, TRUE,
-                                                 TRUE);
+                                                 TRUE, with_src);
     if (hy_query_is_not_empty(query)) {
-        hy_query_filter(query, HY_PKG_ARCH, HY_NEQ, "src");
+        // This inversion is because in the !with_src case, we already
+        // filtered above.
```
Comment (jlebon) [inline libdnf/hy-subject.cpp]: I'm still confused; but then we're always filtering out src pkgs even if `with_src` is `TRUE`, right?

**Review by jlebon — COMMENTED** [review]
I totally missed the fact that this was opened against our fork. Let's open it against upstream libdnf too to get better insight from the dnf side and so we don't have to keep carrying this?

**Comment by dustymabe** [comment]
> I totally missed the fact that this was opened against our fork. Let's open it against upstream libdnf too to get better insight from the dnf side and so we don't have to keep carrying this?

yes please

**Comment by cgwalters** [comment]
I agree, but it's going to take me a while to come back to this as it was just something I was doing on the side and higher priority things preempt it right now.

**Comment by cgwalters** [comment]
Moved to https://github.com/projectatomic/libdnf/pull/2

**Pull request closed without merging**